frontend/device/android/device.lua

```diff
@@ -102,7 +102,7 @@ local Device = Generic:extend{
     importFile = function(path) android.importFile(path) end,
     canShareText = yes,
     doShareText = function(self, text, reason, title, mimetype)
-        android.sendText(text, reason, title, mimetype)
+        android.sendText(text, reason, title or "", mimetype)
```

Comment:
I'd be inclined to think that's more at home on the other side. Pinging @pazos 

https://github.com/koreader/android-luajit-launcher/blob/fda05e7052e2b1b7042d2297b47ef43021dc3122/assets/android.lua#L2476-L2494

Reply:
Yup, but I'm still not sure what happens here. The function @Frenzie linked calls https://github.com/koreader/android-luajit-launcher/blob/master/app/src/main/java/org/koreader/launcher/MainActivity.kt#L625-L641

Where both `reason`, `title`and `mimetype` are nullable strings and properly guarded.

I don't remember anymore but I *think* I tested that `jni.env[0].NewStringUTF(jni.env, nil)` produces a null string when I implemented this. Otherwise the issue you're experiencing could be reproduced if there's no `reason` or no `mimetype`.

If that's the case it is indeed better to address the issue in the lua function linked above for all the three things that might be nil, and change the kotlin function guards to check for empty strings instead of null ones.

Reply:
Thanks @pazos !

So,
1) In android/device.lua:
```
        ...
        android.sendText(text, reason or "", title or "", mimetype or "")
        ...
```
2) and in MainActivity.kt:
```
    override fun sendText(text: String, reason: String?, title: String?, mimetype: String?) {
        if (reason.isEmpty()) {
            reason = ""
        }
        if (title.isEmpty()) {
            title = ""
        }
        if (mimetype.isEmpty()) {
            mimetype = ""
        }
        ...
```

Like this?

Reply:
No, in assets/android.lua:

the tipical

```lua
reason = reason or ""
title = title or ""
mimetype = mimetype or ""
```

And, in MainActivity, do the same guards but instead of checking for null do it for emptry strings.

For instance

```kotlin
 type = mimetype ?: "text/plain"
```

becomes

```koltin
type = if (mimetype.isEmpty()) "text/plain" else mimetype
```

and so on...

Reply:
Thanks! I might get back to this some time later, just don't have the Android device with me all the time anymore.